Formula/anjuta.rb

```diff
@@ -23,12 +23,15 @@ class Anjuta < Formula
   depends_on "hicolor-icon-theme"
   depends_on "libgda"
   depends_on "libxml2"
-  depends_on "python"
+  depends_on "python@3.8"
   depends_on "shared-mime-info"
   depends_on "vala"
   depends_on "vte3"
 
   def install
+    # Required for migration period (3.7 -> 3.8)
+    ENV.prepend_path "PATH", Formula["python@3.8"].opt_bin
```

Comment:
Does brew not already do this? It seems to for library paths at least.

Reply:
Aha, It seems I got confused because `ENV.prepend_path "PATH", Formula["python@3.8"].opt_bin` is required to be added for some tests, for example for `cython` which checks that `cython` module can be imported

Reply:
Hmm, it shouldn't be but I've not tested that.

In fact if Python 3.7 is even installed on the CIs then that's a problem as it's detecting it as a dependency somewhere and thus is potentially being injected by the superenv.

Reply:
> Hmm, it shouldn't be but I've not tested that.

I think it's ok, it allows to test that binaries are available by default (without PATH manipulations from Homebrew users)

> In fact if Python 3.7 is even installed on the CIs then that's a problem as it's detecting it as a dependency somewhere and thus is potentially being injected by the superenv.

Unfortunately, it is for `glib`

Reply:
> I think it's ok, it allows to test that binaries are available by default (without PATH manipulations from Homebrew users)

I think I'm not understanding what's said here. Test that Python 3.8 binaries are available? That should be guaranteed with `depends_on`.

> Unfortunately, it is for `glib`

I'm fine with handling `glib` first and then doing the rest. Of course that might complicate things for any dependents still using 3.7. Perhaps it's worth checking what order these things are prepended to the PATH by the superenv to see if it is consistent behaviour.

Reply:
> I think I'm not understanding what's said here. Test that Python 3.8 binaries are available? That should be guaranteed with `depends_on`.

I mean that PATH shouldn't be prepended by dependencies in `def test` to simulate real user experience for `:keg_only` formulae.

> I'm fine with handling glib first and then doing the rest. Of course that might complicate things for any dependents still using 3.7. Perhaps it's worth checking what order these things are prepended to the PATH by the superenv to see if it is consistent behaviour.

Shouldn't be a problem if `glib` uses `python` only for scripts. We could patch it with something like: 
```ruby
inreplace Dir["#{bin}/{gdbus-codegen,gtester-report,glib-genmarshal,glib-mkenums}"],
  "#!/usr/bin/env python3",
  "#!#{Formula["python@3.8"].opt_bin}/python3"
```

Are you ok with this?

Reply:
> I mean that PATH shouldn't be prepended by dependencies in `def test` to simulate real user experience for `:keg_only` formulae.

For sure, the superenv will handle adding keg only formulae to the path for you.

> Shouldn't be a problem if `glib` uses `python` only for scripts. We could patch it with something like:

I see what you mean now. If we depend on 3.8 then these scripts might not work by default. I suppose that means that a revision bump for `glib` is indeed needed.

We might need to revisit the `glib` subgraph we previously dismissed, for the sake of dependents getting confused at which Python to use. Presumably the superenv doesn't prioritise immediate dependencies over dependencies of dependencies.

Reply:
> For sure, the superenv will handle adding keg only formulae to the path for you.

Yes, it does for `install`, but not for `test`: I've checked it like this:
```ruby
...
  depends_on "python@3.8"

  def install
    xy = Language::Python.major_minor_version "python3"
    ohai "python for `install`", xy
    ...  
  end

  def test
    xy = Language::Python.major_minor_version "python3"
    ohai "python for `test`", xy
    ...  
  end
end
```
And it prints:
```
==> python for `install`
3.8
```

```
==> python for `test`
3.7
```

> I see what you mean now. If we depend on 3.8 then these scripts might not work by default. I suppose that means that a revision bump for `glib` is indeed needed.
> 
> We might need to revisit the `glib` subgraph we previously dismissed, for the sake of dependents getting confused at which Python to use. Presumably the superenv unfortunately doesn't prioritise immediate dependencies over dependencies of dependencies in the PATH.

I don't see a problem here if we do 
```
inreplace Dir["#{bin}/{gdbus-codegen,gtester-report,glib-genmarshal,glib-mkenums}"],
  "#!/usr/bin/env python3",
  "#!#{Formula["python@3.8"].opt_bin}/python3"
```

I maybe lost, could you please elaborate it a bit

Reply:
Oh right I see that stdenv is used in tests rather than superenv. So we'd indeed need to either modify the path or use `#{Formula["python@3.8"].opt_bin}/python3`.

And yes, that's fine for `glib`. I was talking under the presumption that the python@3.8 move is needed now to sort out the superenv but it might not be.